server/server.js

```diff
@@ -1,15 +1,93 @@
 const express = require("express");
+// const fs = require("fs");
+const cors = require("cors");
 const app = express();
-const port = process.env.PORT || 5000;
+const db = require("./config/db");
+const port = process.env.PORT || 7000;
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+app.use(express.json());
+const corsOptions = {
+  origin: "http://localhost:3006", 
+};
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+app.use(cors(corsOptions));
 
-// GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+let videos;
+
+app.get("/", async (req, res) => {
+  const { order } = req.query;
+
+  try {
+    let orderBy = 'DESC'; 
+
+    if (order === 'asc') {
+      orderBy = 'ASC';
+    }
+
+    const queryText = `SELECT * FROM ytvideos ORDER BY ratings ${orderBy}`;
+    const queryResult = await db.query(queryText);
+    const videos = queryResult.rows;
+
+    console.log("Fetched videos from the database:", videos);
+
+    res.json(videos);
+  } catch (error) {
+    console.error("Error fetching data from the database", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
 });
+
+app.post("/", async (req, res) => {
+  const { title, url } = req.body;
+  if (!title || !url) {
+    return res.status(400).json({
+      result: "failure",
+      message: "Both title and url must be provided",
+    });
+  }
+
+  try {
+    // Insert the new video data into your PostgreSQL database
+    const queryText = "INSERT INTO ytvideos (title, url, ratings) VALUES ($1, $2, $3) RETURNING id";
+    const queryValues = [title, url, 7]; 
+    const result = await db.query(queryText, queryValues);
+
+    const id = result.rows[0].id;
+
+    // Return the ID of the newly added video
+    res.status(201).json({ id });
+  } catch (error) {
+    console.error("Error adding video to the database", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+
+
+app.get("/:id", (req, res) => {
+  const id = parseInt(req.params.id);
+  const video = videos.find((video) => video.id === id);
```

Comment:
It looks like this didn't get updated to use the database? Same as the delete, below?